contracts/contracts/eosio.system/src/producer_pay.cpp

```diff
@@ -34,6 +132,35 @@ namespace eosiosystem {
       if( _gstate.total_activated_stake < min_activated_stake )
          return;
 
+      if (schedule_version > _gstate.last_schedule_version) {
+         const auto ct = timestamp.to_time_point();
+         for (const auto p: _gstate.last_schedule) {
+            const auto& prod = _producers.get(p.first.value);
+            const auto max_unpaid_blocks = get_max_possible_produced_blocks(ct, prod.last_max_unpaid_blocks_update, p.first);
```

Comment:
I have some doubts about the `get_max_possible_produced_blocks` function necessity...
Why don't we just save the timestamp of the `last schedule change` and then do something like:

```suggestion
            const auto max_unpaid_blocks = ( ct - min(prod.last_max_unpaid_blocks_update, last_schedule_change_time) ) / producer_count / blocks_per_round;
```

Reply:
@roman-tik I think this can lead us to errors in case when last_max_unpaid_blocks_update or ct is in the middle of production window.

Reply:
what if we always count full production rounds? and never credit rewards for the ongoing round?